node/src/BaseClient.ts

```diff
@@ -1650,22 +1661,21 @@ export class BaseClient {
      * @see {@link https://valkey.io/commands/hset/|valkey.io} for details.
      *
      * @param key - The key of the hash.
-     * @param fieldValueMap - A field-value map consisting of fields and their corresponding values
-     * to be set in the hash stored at the specified key.
+     * @param fieldValueList - A list of field names and their values.
      * @returns The number of fields that were added.
      *
      * @example
      * ```typescript
      * // Example usage of the hset method
-     * const result = await client.hset("my_hash", {"field": "value", "field2": "value2"});
+     * const result = await client.hset("my_hash", [{"field": "field1", "value": "value1"}, {"field": "field2", "value": "value2"}]);
      * console.log(result); // Output: 2 - Indicates that 2 fields were successfully set in the hash "my_hash".
      * ```
      */
     public async hset(
         key: GlideString,
-        fieldValueMap: Record<string, string>,
+        fieldValueList: HashDataType,
```

Comment:
Plaese keep support of old data type as well
```suggestion
        fieldAndValues: HashDataType | Record<string, GlideString>,
```

Reply:
So you can revert changes in tests (but keep in mind you still need to test new API with Buffers)

Reply:
Updated. Have kept some tests with Record type as input and some with HashDataType as input.